hathor/transaction/resources/mempool.py

```diff
@@ -49,18 +50,36 @@ def render_GET(self, request: 'Request') -> bytes:
         set_cors(request, 'GET')
 
         # Get a list of all txs on the mempool
-        assert self.manager.tx_storage.indexes is not None
-        it_mempool = self.manager.tx_storage.indexes.mempool_tips.iter_all(self.manager.tx_storage)
         tx_ids = map(
             # get only tx_ids
             lambda tx: tx.hash_hex,
             # order by timestamp
-            sorted(list(it_mempool), key=lambda tx: tx.timestamp),
+            sorted(list(self._get_from_available_index()), key=lambda tx: tx.timestamp),
         )
         # Only return up to settings.MEMPOOL_API_TX_LIMIT txs per call (default: 100)
         data = {'success': True, 'transactions': list(islice(tx_ids, settings.MEMPOOL_API_TX_LIMIT))}
         return json_dumpb(data)
 
+    def _get_from_available_index(self) -> Iterator[Transaction]:
```

Comment:
Why is that? When would we use one or another? It can get hard to debug bugs when we are not sure which index was used. Isn't it better to allow an external agent to select and let an exception be raised in case of misconfiguration?

Reply:
Do you mean to allow the API user to select which index to use? Not sure I like it, if that's the case

Reply:
The mempool API currently needs the mempool index. Which starting from this PR would only be available if sync-v2 is enabled. But the mempool API is currently being used in the wallet-service.

The behavior of the mempool index can be implemented with the tx-tips index, which is similar but more general. When we switch to sync-v2 it would be preferable to be able to disable the tx-tips index and only use the mempool-index.

But for this API in particular, it is useful for it to work even without the mempool index.

Maybe we can add an `index` parameter, that could accept `index=mempool`, `index=tips` or `index=any`, and maybe we could default to `index=any` for convenience so the wallet-service doesn't have to be updated before this is deployed. What do you think?

Reply:
I ended up implementing just that.

Reply:
Great job! Thanks!